agent/kafka_handler.go

```diff
@@ -17,6 +33,136 @@ type kafkaHandler struct {
 	ctx   kafkaserver2.ConnectionContext
 }
 
+func detectInvalidTopic(name string) error {
+	// Check if the name is empty
+	if len(name) == 0 {
+		return errors.New("the empty string is not allowed")
+	}
+
+	// Check if the name is "." or ".."
+	if name == "." {
+		return errors.New("'.' is not allowed")
+	}
+	if name == ".." {
+		return errors.New("'..' is not allowed")
+	}
+
+	// Check the length of the name
+	if len(name) > maxNameLength {
+		return fmt.Errorf("the length of '%s' is longer than the max allowed length %d", name, maxNameLength)
+	}
+
+	// Check if the name contains only valid characters
+	if !legalChars.MatchString(name) {
+		return fmt.Errorf("'%s' contains one or more characters other than ASCII alphanumerics, '.', '_' and '-'", name)
+	}
+
+	return nil
+}
+
+func (k *kafkaHandler) HandleCreateTopicsRequest(hdr *kafkaprotocol.RequestHeader, req *kafkaprotocol.CreateTopicsRequest, completionFunc func(resp *kafkaprotocol.CreateTopicsResponse) error) error {
+	resp := &kafkaprotocol.CreateTopicsResponse{
+		ThrottleTimeMs: 0,
+		Topics:         make([]kafkaprotocol.CreateTopicsResponseCreatableTopicResult, len(req.Topics)),
+	}
+
+	for tidx, topic := range req.Topics {
+		retentionTime := defaultRetentionTime
+		respConfigs := make([]kafkaprotocol.CreateTopicsResponseCreatableTopicConfigs, len(topic.Configs))
+
+		// Parse custom retention time from topic configs if provided
+		for cidx, config := range topic.Configs {
+			if common.SafeDerefStringPtr(config.Name) == "retention.ms" {
+				retentionMs, err := strconv.Atoi(common.SafeDerefStringPtr(config.Value))
+				if err == nil {
+					retentionTime = time.Duration(retentionMs) * time.Millisecond
+				}
+			}
+
+			respConfigs[cidx] = kafkaprotocol.CreateTopicsResponseCreatableTopicConfigs{
+				Name:  config.Name,
+				Value: config.Value,
+			}
+		}
+
```

Comment:
can we get rid of all this whitespace? We don't really use whitespace much in this project.